src/fe/fe_monomial.C

```diff
@@ -84,7 +82,7 @@ void monomial_nodal_soln(const Elem * elem,
             // u_i = Sum (alpha_i phi_i)
             for (unsigned int i=0; i<n_sf; i++)
               nodal_soln[n] += elem_soln[i] *
-                FEInterface::shape(Dim, p_refined_fe_type, elem, i, refspace_nodes[n]);
+                FEInterface::shape(fe_type, elem, i, refspace_nodes[n]);
```

Comment:
@roystgnr in the course of updating this (and the other FEs) `nodal_soln()` code, I realized that this `FEInerface::shape` call was being made with *both* a "p-refined" `FEType` (i.e. based on `totalorder`) *and* it was being passed an `Elem` pointer. In fe_interface.C, this calls:
```
const Order o = fe_t.order; // this is already totalorder
fe_switch(shape(elem,o,i,p));
```
and that `fe_switch` macros then calls the 4-parameter FE::shape() overload (see fe.h)
```
static OutputShape shape(const Elem * elem,
                         const Order o,
                         const unsigned int i,
                         const Point & p,
                         const bool add_p_level = true);
```
Note that the default value of `add_p_level` is true, so this function is being called with an already-inflated `Order`, and then accounting for the `p_level` *again* internally, for example the 1D monomial instantiation is:
```
template <>
Real FE<1,MONOMIAL>::shape(const Elem * elem,
                           const Order order,
                           const unsigned int i,
                           const Point & p,
                           const bool add_p_level)
{
  libmesh_assert(elem);

  return FE<1,MONOMIAL>::shape(elem->type(), static_cast<Order>(order + add_p_level * elem->p_level()), i, p);
}
```
This definitely seems like a bug to me, but apparently we've never encountered it, possibly because the `n_sf` is still correct and the lower-p DOFs are "nested" within the higher-p DOFs? Or maybe my interpretation of what's happening is just wrong. Either way, it should be fixed in this refactoring.

Reply:
We might not have any p_level() refinement being tested with MONOMIAL elements.  IIRC our test coverage is all for H1 problems with continuous Galerkin.  But you're also right that even solid regression tests (or application code, thankfully) wouldn't have triggered the bug here, since the MONOMIAL shape functions are hierarchic ("nested") with respect to p.  Good catch!